From e97d271460826c89760f33d07661f7a7b2953765 Mon Sep 17 00:00:00 2001
From: Tian Yang <tiany@codeaurora.org>
Date: Tue, 22 Mar 2016 16:17:22 -0500
Subject: [PATCH] skbuff_debug: don't update free stack until the state is
 confirmed to be deactive

If the stack is updated, just the second free stack (another victim)
can be seen. With this change the free stack won't get updated until
the state of the SKB has been confirmed.

The free stack for the second free is still available in the backtrace
of the BUG() output.

Change-Id: I94271f3e03786437c07b23f42c0dcb0591a88836
Signed-off-by: Matthew McClintock <mmcclint@codeaurora.org>
Signed-off-by: Casey Chen <kexinc@codeaurora.org>
---
 net/core/skbuff_debug.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/net/core/skbuff_debug.c
+++ b/net/core/skbuff_debug.c
@@ -176,9 +176,9 @@ inline void skbuff_debugobj_deactivate(s
 
 	obj_state = debug_object_get_state(skb);
 
-	skbuff_debugobj_get_stack(skb->free_addr);
 	if (obj_state == ODEBUG_STATE_ACTIVE) {
 		debug_object_deactivate(skb, &skbuff_debug_descr);
+		skbuff_debugobj_get_stack(skb->free_addr);
 		return;
 	}
 
